### PR TITLE
Feature/navigation

### PR DESCRIPTION
**Pull request by ChiaraDipi**

### CODE REVIEW

File: src/app/panels/text-panel/text-panel.component.ts [hunk 1]
```diff
@@ -14,44 +14,46 @@ import { EVTModelService } from '../../services/evt-model.service';
 export class TextPanelComponent implements OnDestroy {
   @Input() hideEditionLevelSelector: boolean;
 
-  private pid: string;
-  @Input() set pageID(v: string) {
-    this.pid = v;
-    this.pageIDChange.next(this.pid);
+  @Input() set pageID(p: string) {
```
Comment (szenzaro) [inline src/app/panels/text-panel/text-panel.component.ts:17]: Just use currentpageis$

File: src/app/panels/text-panel/text-panel.component.ts [hunk 2]
```diff
@@ -14,44 +14,46 @@ import { EVTModelService } from '../../services/evt-model.service';
...
-    ),
-    this.pages$,
-  ]).pipe(
-    filter(([_, pages]) => !!pages && pages.length > 0),
-    map(([id, pages]) => !id ? pages[0] : pages.find((p) => p.id === id)),
-  ).pipe(
-    distinctUntilChanged((x, y) => JSON.stringify(x) === JSON.stringify(y)),
+  public currentPageId$ = new BehaviorSubject<string>(undefined);
+  public currentPage$ = new BehaviorSubject<Page>(undefined);
+  set currentPage(p: Page) {
```
Comment (szenzaro) [inline src/app/panels/text-panel/text-panel.component.ts:22]: No need of this setter. Just use the subject

File: src/app/panels/text-panel/text-panel.component.ts [hunk 3]
```diff
@@ -14,44 +14,46 @@ import { EVTModelService } from '../../services/evt-model.service';
...
-  @Input() set editionLevelID(el: EditionLevelType) {
-    this.elId = el;
-    this.editionLevelIDChange.next(this.elId);
+  @Input() set editionLevelID(e: EditionLevelType) {
+    this.currentEdLevelId$.next(e);
   }
-  get editionLevelID() { return this.elId; }
-  @Output() editionLevelChange: Observable<EditionLevelType> = this.editionLevelIDChange.pipe(
-    distinctUntilChanged((x, y) => x === y),
+  public currentEdLevelId$ = new BehaviorSubject<EditionLevelType>(undefined);
```
Comment (szenzaro) [inline src/app/panels/text-panel/text-panel.component.ts:34]: Define this from currentlevel$

File: src/app/panels/text-panel/text-panel.component.ts [hunk 4]
```diff
@@ -14,44 +14,46 @@ import { EVTModelService } from '../../services/evt-model.service';
...
-      ),
-      this.pageIDChange,
-    ),
-    this.pages$,
-  ]).pipe(
-    filter(([_, pages]) => !!pages && pages.length > 0),
-    map(([id, pages]) => !id ? pages[0] : pages.find((p) => p.id === id)),
-  ).pipe(
-    distinctUntilChanged((x, y) => JSON.stringify(x) === JSON.stringify(y)),
+  public currentPageId$ = new BehaviorSubject<string>(undefined);
```
Comment (szenzaro) [inline src/app/panels/text-panel/text-panel.component.ts:20]: Define this from currentpages$

File: src/app/panels/text-panel/text-panel.component.ts [hunk 5]
```diff
@@ -14,44 +14,46 @@ import { EVTModelService } from '../../services/evt-model.service';
...
-    this.editionLevelIDChange.next(this.elId);
+  @Input() set editionLevelID(e: EditionLevelType) {
+    this.currentEdLevelId$.next(e);
   }
-  get editionLevelID() { return this.elId; }
-  @Output() editionLevelChange: Observable<EditionLevelType> = this.editionLevelIDChange.pipe(
-    distinctUntilChanged((x, y) => x === y),
+  public currentEdLevelId$ = new BehaviorSubject<EditionLevelType>(undefined);
+  public currentEdLevel$ = new BehaviorSubject<EditionLevel>(undefined);
+  set currentEdLevel(e: EditionLevel) {
```
Comment (szenzaro) [inline src/app/panels/text-panel/text-panel.component.ts:36]: No need of this setter

File: src/app/services/evt-status.service.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+export type URLParamsKeys = 'd' | 'p' | 'el' | 'ws' | 'vs';
+export type URLParams = { [T in URLParamsKeys]: string };
+
+@Injectable({
+    providedIn: 'root',
+})
+export class EVTStatusService {
+    public availableEditionLevels = AppConfig.evtSettings.edition.availableEditionLevels?.filter((e => !e.disabled)) || [];
+    get defaultEditionLevel(): EditionLevelType {
```
Comment (szenzaro) [inline src/app/services/evt-status.service.ts:18]: Rename defauleditonlevelID

File: src/app/view-modes/collation/collation.component.ts [hunk 7]
```diff
@@ -21,20 +21,21 @@ export class CollationComponent implements OnInit, OnDestroy {
...
+    private evtStatusService: EVTStatusService,
   ) {
   }
 
   ngOnInit() {
     this.initGridster();
-    this.initPageAndWitnesses();
+  }
+
+  handlePageChange(selectedPage: Page) {
```
Comment (szenzaro) [inline src/app/view-modes/collation/collation.component.ts:37]: Rename "changepage"?

File: src/app/view-modes/text-text/text-text.component.ts [hunk 8]
```diff
@@ -1,50 +1,73 @@
...
+          edLvl1 = this.evtStatusService.availableEditionLevels.filter(e => e.id !== edLvl2.id)[0];
+        }
+      }
+      this.evtStatusService.updateEditionLevels$.next([edLvl1?.id, edLvl2?.id]);
+    });
   }
 
-  handleEditionLevelChange(editionLevel: EditionLevelType, paramName: string) {
-    this.editionLevel.handleEditionLevelChange(this.route, editionLevel, paramName);
+  handlePageChange(selectedPage: Page) {
```
Comment (szenzaro) [inline src/app/view-modes/text-text/text-text.component.ts:60]: Rename pagechange?

File: src/app/view-modes/text-versions/text-versions.component.ts [hunk 9]
```diff
@@ -36,8 +37,12 @@ export class TextVersionsComponent implements OnInit {
     this.initPageAndVersions();
   }
 
-  handleEditionLevelChange(editionLevel: EditionLevelType) {
-    this.editionLevel.handleEditionLevelChange(this.route, editionLevel, 'edLvl');
+  handlePageChange(selectedPage: Page) {
+    this.evtStatusService.updatePage$.next(selectedPage);
+  }
+
+  handleEditionLevelChange(editionLevel: EditionLevel) {
```
Comment (szenzaro) [inline src/app/view-modes/text-versions/text-versions.component.ts:44]: Rename changeeditionlevel?

**Comment by szenzaro** [comment]
Missing changelog update

File: src/app/view-modes/collation/collation.component.ts (hunk 7 above)
Comment (ChiaraDipi) [reply src/app/view-modes/collation/collation.component.ts:37]: you mean renaming completely the method in `changePage` or just invert words and have `handleChangePage`?

File: src/app/components/page-selector/page-selector.component.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+@Component({
+  selector: 'evt-page-selector',
+  templateUrl: './page-selector.component.html',
+  styleUrls: ['./page-selector.component.scss'],
+})
+export class PageSelectorComponent {
+  public pages$ = this.evtModelService.pages$;
+
+  private _pageID: string;
+  @Input() set page(p: string) {
```
Comment (szenzaro) [inline src/app/components/page-selector/page-selector.component.ts:16]: is this a page or a pageID?

File: src/app/services/evt-status.service.ts [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+        this.updateDocument$,
+    );
+    public currentPage$ = combineLatest([
+        merge(
+            this.route.queryParams.pipe(map((params: URLParams) => params.p)),
+            this.updatePage$.pipe(map(p => p.id)),
+        ),
+        this.evtModelService.pages$,
+    ]).pipe(
+        filter(([_, pages]) => !!pages && pages.length > 0),
```
Comment (szenzaro) [inline src/app/services/evt-status.service.ts:60]: Put this filter after the `this.evtModelService.pages$`

File: src/app/view-modes/reading-text/reading-text.component.ts [hunk 12]
```diff
@@ -15,13 +14,12 @@ import { EditionLevelService } from '../../services/edition-level.service';
 export class ReadingTextComponent implements OnInit, OnDestroy {
   public layoutOptions: GridsterConfig = {};
   public textPanelItem: GridsterItem = { cols: 1, rows: 1, y: 0, x: 0 };
-  public currentPage = this.route.params.pipe(
-    map((params) => params.page),
+  public currentPage$ = this.evtStatusService.currentStatus$.pipe(
```
Comment (szenzaro) [inline src/app/view-modes/reading-text/reading-text.component.ts:17]: currentPageID ?

File: src/app/view-modes/text-text/text-text.component.ts [hunk 13]
```diff
@@ -1,50 +1,73 @@
...
-  private defaultEdLvl2 = AppConfig.evtSettings.edition.availableEditionLevels?.filter((e => !e.disabled))[1];
-
-  public currentEditionLevel = this.route.params.pipe(
-    map((params) => params.edLvl ?? this.defaultEdLvl1?.id),
-    distinctUntilChanged((x, y) => x === y),
-  );
-  public currentEdLvlPanel1 = this.route.params.pipe(
-    map((params) => params.edLvl ?? this.defaultEdLvl1?.id),
-    distinctUntilChanged((x, y) => x === y),
+  public currentPage$ = this.evtStatusService.currentStatus$.pipe(
```
Comment (szenzaro) [inline src/app/view-modes/text-text/text-text.component.ts:20]: currentPageID ?

File: src/app/view-modes/text-versions/text-versions.component.ts [hunk 14]
```diff
@@ -19,15 +19,16 @@ export class TextVersionsComponent implements OnInit {
...
-  private defaultEditionLevel = AppConfig.evtSettings.edition.availableEditionLevels?.filter((e => !e.disabled))[0];
-  public currentEditionLevel = this.route.params.pipe(
-    map((params) => params.edLvl ?? this.defaultEditionLevel?.id),
-    distinctUntilChanged((x, y) => x === y),
+  public currentPage$ = this.evtStatusService.currentStatus$.pipe(
+    map(({ page }) => page.id),
+  );
+
+  public currentEditionLevel$ = this.evtStatusService.currentStatus$.pipe(
+    map(({ editionLevels }) => editionLevels[0]),
```
Comment (szenzaro) [inline src/app/view-modes/text-versions/text-versions.component.ts:27]: Can `editionLevels` have zero length?

File: src/app/components/page-selector/page-selector.component.ts (hunk 10 above)
Comment (ChiaraDipi) [reply src/app/components/page-selector/page-selector.component.ts:16]: `pageID`. I'll change it

File: src/app/view-modes/text-versions/text-versions.component.ts (hunk 14 above)
Comment (ChiaraDipi) [reply src/app/view-modes/text-versions/text-versions.component.ts:27]: Yes, but if zero length, first element is `undefined`. 
If you prefer I can rewrite this line like this:
```
 map(({ editionLevels }) => editionLevels.length > 0 ? editionLevels[0] : undefined),
```
or 
```
 map(({ editionLevels }) => editionLevels[0] ?? undefined),
```

**Pull request merged**